tools/pythonpkg/tests/fast/api/test_read_csv.py

```diff
@@ -185,4 +186,32 @@ def test_filename(self, duckdb_cursor):
 		df = rel.df()
 		column_names = list(df.columns.values)
 		# The filename is included in the returned columns
-		assert 'filename' in column_names
\ No newline at end of file
+		assert 'filename' in column_names
+
+	def test_read_filelike(self, duckdb_cursor):
+		string = StringIO("c1,c2,c3\na,b,c")
+		res = duckdb_cursor.read_csv(string, header=True).fetchall()
```

Comment:
Could we add some more tests?

* BytesIO
* Other objects that have a read() method maybe?
* What if we pass in a non-readable object (e.g. an integer)
* What if we pass in "None"
* What if read() throws an exception? (perhaps we can make our own class that has a read method that throws an exception?)
* Can we somehow verify the object gets properly deleted? For example, we could make our own class that updates some global in the destructor, e.g. something like this:

```py
global_counter = 0
class MyObject:
  def __init__(self):
    global global_counter
    global_counter += 1
  def __del__(self):
    global global_counter
    global_counter -= 1
```
```py
>>> print(global_counter)
0
>>> obj = MyObject()
>>> print(global_counter)
1
>>> del obj
>>> print(global_counter)
0

```